packages/synapse-react-client/src/synapse-queries/wiki/useWiki.ts

```diff
@@ -18,34 +24,100 @@ import {
   UseQueryOptions,
 } from '@tanstack/react-query'
 
+function getRootWikiPageKeyQueryOptions(
+  objectType: ObjectType,
+  objectId: string,
+  synapseContext: SynapseContextType,
+): UseQueryOptions<WikiPageKey | null, SynapseClientError> {
+  const { keyFactory, accessToken } = synapseContext
+  return {
+    queryKey: keyFactory.getRootWikiPageKeyQueryKey(objectType, objectId),
+    queryFn: () => getRootWikiPageKey(accessToken, objectType, objectId),
+  }
+}
+
 export const useGetRootWikiPageKey = (
   ownerObjectType: ObjectType,
   ownerObjectId: string,
   options?: Partial<UseQueryOptions<WikiPageKey | null, SynapseClientError>>,
 ) => {
-  const { accessToken, keyFactory } = useSynapseContext()
+  const synapseContext = useSynapseContext()
 
   return useQuery({
     ...options,
-    queryKey: keyFactory.getRootWikiPageKeyQueryKey(
+    ...getRootWikiPageKeyQueryOptions(
       ownerObjectType,
       ownerObjectId,
+      synapseContext,
     ),
-    queryFn: () =>
-      getRootWikiPageKey(accessToken, ownerObjectType, ownerObjectId),
   })
 }
 
 export const useGetWikiPage = (
   wikiPageKey: WikiPageKey,
   options?: Partial<UseQueryOptions<WikiPage, SynapseClientError>>,
 ) => {
-  const { accessToken, keyFactory } = useSynapseContext()
+  const synapseContext = useSynapseContext()
+  const { accessToken, keyFactory } = synapseContext
+  const queryClient = useQueryClient()
 
   return useQuery({
     ...options,
     queryKey: keyFactory.getWikiPageQueryKey(wikiPageKey),
-    queryFn: () => getWikiPage(accessToken, wikiPageKey),
+    queryFn: async () => {
+      let wikiPageId = wikiPageKey.wikiPageId
+      if (!wikiPageId) {
+        const rootWikiPageKey = await queryClient.fetchQuery(
+          getRootWikiPageKeyQueryOptions(
+            wikiPageKey.ownerObjectType,
+            wikiPageKey.ownerObjectId,
+            synapseContext,
+          ),
+        )
```

Comment:
Find the wikiPageId if it was null -- we must do this for non-entities (e.g. attempting to GET a wiki for an access requirement without the wikiPageId is not supported by the backend)